db/version_set.cc

```diff
@@ -5145,7 +5149,7 @@ Status VersionSet::DumpManifest(Options& options, std::string& dscname,
       Version* v = new Version(cfd, this, file_options_,
                                *cfd->GetLatestMutableCFOptions(),
                                current_version_number_++);
-      builder->SaveTo(v->storage_info());
+      s = builder->SaveTo(v->storage_info());
```

Comment:
Since this happens in a loop, we probably should break out of the loop once an error is detected.

Reply:
This is for VersionSet::DumpManifest(). It depends on whether we want ldb still to dump the contents of manfiest file if a corruption is detected in the middle. I think it can go both way but now I incline to go with a way that does minimum change to the tool's behavior.